test/src/main/java/org/axonframework/test/aggregate/TestExecutor.java

```diff
@@ -119,20 +119,52 @@
     Instant currentTime();
 
     /**
-     * Simulates time shifts in the current given state. This can be useful when the time between given events is of
-     * importance.
+     * Simulates the time elapsing in the current given state using a {@link Duration} as the unit of time. This can be
+     * useful when the time between given events is of importance, for example when leveraging the
+     * {@link org.axonframework.deadline.DeadlineManager} to schedule deadlines in the context of a given Aggregate.
      *
-     * @param elapsedTime The amount of time that will elapse
-     * @return a ResultValidator that can be used to validate the resulting actions of the command execution
+     * @param elapsedTime a {@link Duration} specifying the amount of time that will elapse
+     * @return a {@link ResultValidator} that can be used to validate the resulting actions of the command execution
+     *
+     * @deprecated in favor of {@link #whenThenTimeElapses(Duration)}. This function incorrectly suggests you can
+     * proceed with other operations after calling it, which is made impossible due to the {@link ResultValidator}
+     * return type
      */
+    @Deprecated
     ResultValidator andThenTimeElapses(Duration elapsedTime);
 
     /**
-     * Simulates time shifts in the current given state. This can be useful when the time between given events is of
-     * importance.
+     * Simulates the time elapsing in the current given state using a {@link Duration} as the unit of time. This can be
+     * useful when the time between given events is of importance, for example when leveraging the
+     * {@link org.axonframework.deadline.DeadlineManager} to schedule deadlines in the context of a given Aggregate.
      *
-     * @param newDateTime The time to advance the clock to
-     * @return a ResultValidator that can be used to validate the resulting actions of the command execution
+     * @param elapsedTime a {@link Duration} specifying the amount of time that will elapse
+     * @return a {@link ResultValidator} that can be used to validate the resulting actions of the command execution
+     */
+    ResultValidator whenThenTimeElapses(Duration elapsedTime);
```

Comment:
I'd make this a default message and invoke `andThenTimeElapses`.

Reply:
![outstanding-move](https://user-images.githubusercontent.com/4687743/54022182-34e58b00-4192-11e9-9304-b17634e508d9.jpg)